src/vm/eventpipefile.cpp

```diff
@@ -138,8 +124,38 @@ EventPipeFile::EventPipeFile(StreamWriter *pStreamWriter, EventPipeSerialization
     QueryPerformanceCounter(&m_lastSortedTimestamp);
 #endif
 
+
+}
+
+void EventPipeFile::InitializeFile()
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    }
+    CONTRACTL_END;
+
+    bool fSuccess = true;
+    if (m_format >= EventPipeSerializationFormat::NetTraceV4)
+    {
+        const char* pHeader = "Nettrace";
+        uint32_t bytesWritten = 0;
+        fSuccess = m_pStreamWriter->Write(pHeader, 8, bytesWritten) && bytesWritten == 8;
+    }
+    if (fSuccess)
+    {
+        // Create the file stream and write the FastSerialization header.
+        m_pSerializer = new FastSerializer(m_pStreamWriter);
+    }
+    else
+    {
+        m_pSerializer = nullptr;
+    }
```

Comment:
If `m_pSerializer` is set to `nullptr` line 157 below can throw an AV.  Should you move lines 157-158 under `if (fSuccess)` block and set `m_isInitialized = false;` in the else block?

Reply:
To Sung's comment, the new operator will throw rather than returning a nullptr. There is a new (nothrow) variant for code that does its own explicit null check.